src/lib/operators/aggregate_hash.cpp

```diff
@@ -688,123 +688,102 @@ template <typename ColumnDataType, typename AggregateType, AggregateFunction fun
 std::enable_if_t<func == AggregateFunction::Min || func == AggregateFunction::Max || func == AggregateFunction::Sum ||
                      func == AggregateFunction::Any,
                  void>
-write_aggregate_values(std::shared_ptr<ValueSegment<AggregateType>> segment,
+write_aggregate_values(pmr_vector<AggregateType>& values, pmr_vector<bool>& null_values,
```

Comment:
Changes here are necessary because write_aggregate_values previously modified the values of an already existing ValueSegment.